activiti-core/activiti-api-impl/activiti-api-runtime-shared-impl/src/main/java/org/activiti/runtime/api/event/internal/VariableCreatedListenerDelegate.java

```diff
@@ -33,32 +37,47 @@ public class VariableCreatedListenerDelegate implements ActivitiEventListener {
 
     private final VariableEventFilter variableEventFilter;
 
+    private final ProcessExtensionService processExtensionService;
+
+
     public VariableCreatedListenerDelegate(
         List<VariableEventListener<VariableCreatedEvent>> listeners,
         ToVariableCreatedConverter converter,
-        VariableEventFilter variableEventFilter) {
+        VariableEventFilter variableEventFilter, ProcessExtensionService processExtensionService) {
         this.listeners = listeners;
         this.converter = converter;
         this.variableEventFilter = variableEventFilter;
+        this.processExtensionService = processExtensionService;
     }
 
     @Override
     public void onEvent(ActivitiEvent event) {
         if (event instanceof ActivitiVariableEvent) {
             ActivitiVariableEvent internalEvent = (ActivitiVariableEvent) event;
+            Extension extension = processExtensionService.getExtensionsForId(internalEvent.getProcessDefinitionId());
             if (variableEventFilter.shouldEmmitEvent(internalEvent)) {
-                converter.from(internalEvent)
-                    .ifPresent(convertedEvent -> {
-                        if (listeners != null) {
-                            for (VariableEventListener<VariableCreatedEvent> listener : listeners) {
-                                listener.onEvent(convertedEvent);
-                            }
-                        }
-                    });
+                VariableDefinition extensionPropertyByName=null;
+                if(extension!=null) {
+                    extensionPropertyByName = extension.getPropertyByName(internalEvent.getVariableName());
+                }
+                boolean isEphemeral = extension != null && (extensionPropertyByName!=null && extensionPropertyByName.isEphemeral());
```

Comment:
I think you can add a helper method directly in processExtensionService so you can call directly `processExtensionService .hasEphemeralVariable(internalEvent.getProcessDefinitionId(), internalEvent.getVariableName())`. This will also allow you to reuse it for update and delete variables events